### PR TITLE
Add /craic:reflect slash command (#9)

### PR DESCRIPTION
**Pull request by dni138**
## Summary

- Implements the `/craic:reflect` slash command (`commands/craic-reflect.md`), closing #9
- Fixes Docker permission errors in `team-api/Dockerfile` for the system user (`adduser --system`)

## What /craic:reflect does

Seven-step session mining flow:
1. Synthesises a dense session summary (APIs, errors, workarounds, dead ends)
2. Calls `craic_reflect` MCP tool with the context (forward-compatible with when the stub gets real server-side logic)
3. Independently identifies candidate knowledge units using LLM reasoning — looking for undocumented API behaviour, non-obvious workarounds, condition-specific config, multi-attempt error resolutions, version incompatibilities, novel patterns
4. Presents candidates as a numbered list with domain tags and relevance score
5. Accepts approve / skip / edit per candidate (or "all" / "none")
6. Calls `craic_propose` for each approved candidate and confirms with the stored ID
7. Displays a `## Session Reflect Complete` summary

## Dockerfile fix

`adduser --system` sets `HOME=/nonexistent`. Any tool that creates a cache under `~` at runtime (including `uv`) fails with Permission denied. Fixed by:
- `chown -R app:app /app` so the app user owns the venv
- `CMD ["/app/.venv/bin/craic-team-api"]` — invokes the venv binary directly, bypassing uv at runtime entirely

## Test plan

- [x] `craic_reflect` MCP stub called and handled correctly (empty candidates → LLM reasoning takes over)
- [x] Candidate identification from session context
- [x] `craic_propose` end-to-end: local store + team API sync (HTTP 201 confirmed)
- [x] Empty session edge case — correct "no candidates" message
- [x] "all" approval edge case — both candidates stored and synced to team store
- [x] Existing `test_server.py` tests for `craic_reflect` stub pass unchanged

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: commands/craic-reflect.md [hunk 1]
```diff
@@ -1,3 +1,152 @@
...
+- Optionally: **language**, **framework**, **pattern** if relevant.
+
+If the session contained no events meeting the above criteria, note that explicitly and stop after this step.
+
+### Step 4 — Present candidates to the user
+
+Open with:
+
+```
+I identified {N} potential learning{s} from this session worth sharing with the commons.
```
[the remark this answers is not in the document]
Comment (dni138) [reply commands/craic-reflect.md]: Fixed in f80f965 — replaced with "learning candidates" which is both unambiguous and avoids any template-substitution confusion.

File: commands/craic-reflect.md [hunk 2]
```diff
@@ -1,3 +1,152 @@
...
+- **detail** — two to four sentences explaining the context and why this behaviour exists or matters.
+- **action** — a concrete instruction on what to do (start with an imperative verb).
+- **domain** — two to five lowercase domain tags (e.g. `["api", "stripe", "rate-limiting"]`).
+- **estimated_relevance** — a float between 0.0 and 1.0:
+  - 0.8–1.0: broadly applicable across many languages, frameworks, or teams.
+  - 0.5–0.8: applicable to a specific ecosystem or toolchain.
+  - 0.2–0.5: applicable only under narrow conditions.
+- Optionally: **language**, **framework**, **pattern** if relevant.
+
+If the session contained no events meeting the above criteria, note that explicitly and stop after this step.
```
[the remark this answers is not in the document]
Comment (dni138) [reply commands/craic-reflect.md]: Fixed in f80f965 — Step 3 now reads "skip Steps 4–6 and follow the 'no candidates' instruction in Step 7" so there's a single clear termination path.
